modules/core/src/main/java/org/apache/ignite/spi/IgniteSpiOperationTimeoutHelper.java

```diff
@@ -27,7 +27,11 @@
  *
  * A new instance of the class should be created for every complex network based operations that usually consists of
  * request and response parts.
+ *
+ * Please use {@link ExponentialBackoffSpiTimeout}.
+ *
  */
+@Deprecated
```

Comment:
Deprecation is wrong because it's used in discovery SPI.

Reply:
IgniteSpiOperationTimeoutHelper is poorly class but without deprecation annotation we might leave it as is, i agree that we can remove deprecated annotation and create jira to refactor discovery to use ExponentialBackoffTimeout.